python_modules/dagster/dagster/_core/definitions/external_asset.py

```diff
@@ -125,44 +126,60 @@ def _external_assets_def(context: AssetExecutionContext) -> None:
     return assets_defs
 
 
+# SYSTEM_METADATA_KEY_AUTO_OBSERVE_INTERVAL_MINUTES lives on the metadata of
+# external assets resulting from a source asset conversion. It contains the
+# `auto_observe_interval_minutes` value from the source asset and is consulted
+# in the auto-materialize daemon. It should eventually be eliminated in favor
+# of an implementation of `auto_observe_interval_minutes` in terms of
+# `AutoMaterializeRule`.
+SYSTEM_METADATA_KEY_AUTO_OBSERVE_INTERVAL_MINUTES = "dagster/auto_observe_interval_minutes"
+
+
 def create_external_asset_from_source_asset(source_asset: SourceAsset) -> AssetsDefinition:
-    check.invariant(
-        source_asset.auto_observe_interval_minutes is None,
-        "Automatically observed external assets not supported yet: auto_observe_interval_minutes"
-        " should be None",
-    )
-
-    kwargs = {
-        "key": source_asset.key,
-        "metadata": source_asset.metadata,
-        "group_name": source_asset.group_name,
-        "description": source_asset.description,
-        "partitions_def": source_asset.partitions_def,
-        "_execution_type": (
-            AssetExecutionType.UNEXECUTABLE
-            if source_asset.observe_fn is None
-            else AssetExecutionType.OBSERVATION
+    observe_interval = source_asset.auto_observe_interval_minutes
+    metadata = {
+        **source_asset.raw_metadata,
+        **(
+            {SYSTEM_METADATA_KEY_AUTO_OBSERVE_INTERVAL_MINUTES: observe_interval}
+            if observe_interval
+            else {}
```

Comment:
I find it weird that you went through all the effort to remove the execution type from metadata in the upstream diff but then you are readding metadata here. I still think stashing in metadata is the right move for now.